Cleanup

goober48 opened the pull request:
resolves https://github.com/AEMBOT/FRC_2025/issues/54

notpro-ek committed:
added the intake canrange detector. has a boolean for vertical corals intaked or not intaked and a function to get coral distance from the center of the intake which will return both negative and positive values. Looking to add values to the driver assist to align with reef.

notpro-ek committed:
UNTESTED new driver assist based on the offset of vertical coral location. i hope this works

notpro-ek committed:
Save off changes. It needs to be the final robot position, not the apriltag location.

(Use PositionConstants.reefX and PositionConstants.reefY)

TheFlameFish committed:
Sync with main and slight refactoring

TheFlameFish committed:
Remove unused methods

notpro-ek committed:
fixed refactoring and cleaned up files. Coral offset is now giving the proper robot pose and not the april tag pose, for AprilTag 18 and AprilTag 21, the easiest ones to calculate by hand. Rest of the apriltags might need to be tested and calculated by hand, but the pose values that are being given look to be right. Also build changes to stop indent deprecated text.

notpro-ek committed:
Merge branch 'intake-distance-sensor' of https://github.com/AEMBOT/FRC_2025 into intake-distance-sensor

notpro-ek committed:
Fixed broken test target pose code after merge. Tests are still working. Fixed documentation.

TheFlameFish committed:
Round offset tests

TheFlameFish committed:
Re-add bindings

TheFlameFish committed:
ignore coral position when going to L1

TheFlameFish committed:
Slight documentation & refactoring

TheFlameFish committed:
Remove unneeded method

TheFlameFish committed:
Use assertions with Pose2ds instead of converting to arrays

TheFlameFish committed:
Revert out-of-scope build.gradle change

TheFlameFish committed:
Minor formatting, docs, and refactor

TheFlameFish committed:
Simplify getting nearest tag

TheFlameFish committed:
Change ReefTargetsTest to use new method of getting nearest apriltag, minor refactor

TheFlameFish committed:
give nearest tag tests a small amount of tolerance

TheFlameFish committed:
Remove now unused method

TheFlameFish committed:
Switch delta to theta (oops)

TheFlameFish committed:
Make red test tolerance consistent with blue

TheFlameFish committed:
Remove assertPoseEquals because floating point errors render it non-useful

TheFlameFish committed:
Properly revert spotless indent changes

TheFlameFish committed:
Leave source once we have coral in auto

TheFlameFish committed:
Spotless

TheFlameFish committed:
Make tolerance shared between all reef test methods

notpro-ek committed:
Make the function to check for having a game piece work for horizontal corals as well. Still need to fix the getGamePieceLocation to work with horizontal corals as well. WIP.

notpro-ek committed:
Code to detect horizontal coral pieces.

notpro-ek committed:
build gradle + spotless

nolanhergert committed:
* Need to subtract a positive offset, instead of negative offset
* Clean up unneeded changes

nolanhergert committed:
Undo spotless changes for a separate PR

nolanhergert committed:
simplified try catch statements

nolanhergert committed:
Merge remote-tracking branch 'origin/main' into refactor-unit-tests

goober48 committed:
Test commit

goober48 committed:
cleaned up deprecated functions

TheFlameFish committed:
Minor tweaks made at Chicken Coop (#58)

* Move reef poses closer

* Log swerve motor temps

* Remove delay in auto intake and place

* Log alignment error

* Adjust reef targets units test

Ace5tar committed:
Stow position and arm ordering (#46)

* added stow position and adjusted controls

* controls for stow position

* merge with main

* added correct arm ordering and correct stow position usage

* added stow to named commands for use during auto

* changes for drive controls and testing on nautilus

* Tests on nautilus

* minor arm changes and HUGE speed increase for pivot and wrist

* chages made at chicken coop

* reordered wirst motion magic to be accurate to actual speeds

* removed incorrect addition to intake volts logging

* Change instances of armToReef to armToReefSafely

* Spotless did smth

* Fix recursive oopsie

---------

Co-authored-by: Ash <114826427+TheFlameFish@users.noreply.github.com>
Co-authored-by: FlameFish <mccoa499@gmail.com>

notpro-ek committed:
easier on the eyes. tiny line of code fix for pivot. working on understanding vision and drive branches, but for now i don't really understand it, so no changes to those branches.

notpro-ek committed:
Cleaned up constants. Cleaned up unused functions.
Odometry deprecated fix. (This is not tested and we don't know if it works.)

notpro-ek committed:
More constant changes.

TODO: check odometry changes and check if the deprecated method fix works.

notpro-ek committed:
Merge branch 'main' into cleanup

notpro-ek committed:
Build + Spotless

notpro-ek committed:
Merge error fixes

nolanhergert commented on src/main/java/frc/robot/constants/WristConstants.java:
```diff
@@ -40,40 +29,9 @@ public class WristConstants {
   public static final double ALLOWED_DEVIANCE = 6;
 
-  /** */
-  public static final Translation3d TRANSLATION_FROM_ROBOT = new Translation3d(-0.2, 0, 0.255);
```
Don't you need this constant for sim? (and eventually optimal inverse kinematics solving)

notpro-ek replied:
It is not used in sim, and in the future, if we use inverse kinematics, I don't think we would use this exact variable type. I'm uncertain of what type of constant we would want, like perhaps we would want a vector, or a translation2d, instead of a translation3d. Currently, I do not think this is needed.

nolanhergert commented on src/main/java/frc/robot/constants/PivotConstants.java:
```diff
@@ -46,53 +35,9 @@ public class PivotConstants {
   public static final double ALLOWED_DEVIANCE = 1.15;
 
-  /** */
-  public static final Translation3d TRANSLATION_FROM_ROBOT = new Translation3d(-0.2, 0, 0.255);
```
Don't you need this constant for sim? (and eventually optimal inverse kinematics solving)

nolanhergert commented on src/main/java/frc/robot/constants/ElevatorConstants.java:
```diff
@@ -36,66 +25,9 @@ public class ElevatorConstants {
   public static final double ALLOWED_DEVIANCE = 0.01;
 
-  /** */
-  public static final Translation3d TRANSLATION_FROM_ROBOT = new Translation3d(-0.2, 0, 0.255);
```
Don't you need this constant for sim? (and eventually optimal inverse kinematics solving)

nolanhergert commented on build.gradle:
```diff
@@ -179,7 +179,7 @@ spotless {
             exclude "**/build/**", "**/build-*/**"
         }
         trimTrailingWhitespace()
-        indentWithSpaces(2)
```
Is indentWithSpaces deprecated? Can you explain why this (and some of the other changes) are required in the PR description?

nolanhergert replied:
Ahh, I already wrote it in the description of the issue, nevermind! Resolving.

nolanhergert commented on src/main/java/frc/robot/subsystems/elevator/ElevatorIOReal.java:
```diff
@@ -86,14 +74,11 @@ public void updateInputs(ElevatorIOInputs inputs) {
   public void limitHeight(double pivotAngle) {
-    ++i;
+
     Logger.recordOutput("i", pivotAngle);
```
I am quite certain you can remove this logger output too.

notpro-ek replied:
I think you're right.

nolanhergert reviewed (commented):
Great work! I think you (and the other reviewers) will be happier if this is split into several PRs/branches so they are more coherent.

Apparently you can just run `git checkout origin/main <filename>` to reset a particular file back to main. But no rush, we can do it in person next week.

notpro-ek committed:
Documentation fixes. A little bit of clean up here and there.

We seem to have been logging elevator velocity in elevator io real improperly and should have been getting wrong values from it?

Should be fixed now though.

Hoping to fix documentation of code before new programmers come in.

notpro-ek committed:
remove non-needed logger recording in limitHeight function
